subiquity/common/apidef.py

```diff
@@ -403,6 +405,11 @@ def GET(user_id: str) -> SSHFetchIdResponse: ...
     class integrity:
         def GET() -> CasperMd5Results: ...
 
+    class active_directory:
+        def GET() -> Optional[ADConnectionInfo]: ...
+        def POST(data: Payload[ADConnectionInfo]) \
+            -> List[ADValidationResult]: ...
```

Comment:
In what scenario would we see multiple results?

Reply:
Multiple errors. Since this is meant to validate the whole structure at once we may have a combination of errors on username and hostname, for instance. I was in doubt about the success case: empty list versus OK value. My take was in favor of explicitness.

Reply:
Hmm, I don't know how username validation will work if the hostname is wrong, but I'm in favor of returning all the info we have.  I suggest a comment as this sort of multiple errors return is a bit uncommon.

Reply:
That validation is just matching regexes. I'm placing that validation in the server to avoid code duplication on the clients.